lib/internal/process/promises.js

```diff
@@ -26,6 +33,7 @@ function setupPromises(scheduleMicrotasks) {
     if (hasBeenNotified !== undefined) {
       hasBeenNotifiedProperty.delete(promise);
       if (hasBeenNotified === true) {
+        promiseInternals.untrackPromise(promise);
```

Comment:
In Blink, we track at most 1000 promises: https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/Source/bindings/core/v8/RejectedPromises.cpp&l=26 - the assumption is that an unhandled rejection is a bug in the app, and there's only so and so much value in tracking more than a handful of bugs.